pinot-spi/src/main/java/org/apache/pinot/spi/data/Schema.java

```diff
@@ -59,13 +65,28 @@
   private static final Logger LOGGER = LoggerFactory.getLogger(Schema.class);
 
   private String _schemaName;
+
+  // upsert related config
+  // primary key refers to the column name that is the primary key of this upsert table
+  private String _primaryKey;
```

Comment:
what if the primary key is made up of two columns?

Reply:
or multiple

Reply:
right now we don't have support for multiple primary keys. Do you think it will be an important feature to add multiple primary key support right now or you think it is fine to postpone it to later diff?

Reply:
Better to design it for multiple, even if the implementation can handle only one primary key

Reply:
> don't have support for multiple primary keys

There will be some cases where multiple keys make sense right? Like in clickstream: `userId and timestamp`

Reply:
@kishoreg @ChethanUK I have changed the primary key in schema config to be a list instead of a single string. For now we are still performing check to ensure that primary key will be one and only one. However we should be able to change that once we have mulitple-primary-key support in place